tfm_ns_import.yaml

```diff
@@ -18,22 +18,22 @@
     "mbed-os": {
         "ARM_MUSCA_S1": [
             {
-                "src": "image_macros_preprocessed_ns.c",
-                "dst": "targets/TARGET_ARM_SSG/TARGET_MUSCA_S1/partition/image_macros_preprocessed_ns.c"
+                "src": "bl2/ext/mcuboot/CMakeFiles/signing_layout_ns.dir/signing_layout_ns.o",
```

Comment:
Shouldn't that be ".c"?

Reply:
It is `*.o` in tf-m's build output. So, this is actually a text file containing a C-style enum, but actually parsed by the signing tool (so not a "real" source file).
This is related to the discussion here: https://github.com/LDong-Arm/mbed-os/commit/18f4b198866e65c4d44da267f6d788d35cb8297d#r46472131
I have no strong preference over the file extension, but ".o" is the original one and might be less confusing.

Reply:
But it's been called `*.c` since previous versions of TF-M integration, so I'm really fine with both

Reply:
The TF-M v1.1 had `image_macros_preprocessed_ns.c` file with preprocessed macro values. 
In the new TF-M v1.2 , the image_macros_preprocessed_ns.c is renamed to `signing_layout_ns.c` which doesn't have preprocessed macros due to new build system, therefore using preprocessed `signing_layout_ns.o` from build folder.
On the other hand, Mbed CLI 1 doesn't understand `.o` unless major changes are made to the script and build system, therefore renaming to `.c` works.

I'll document this in the commit and open an issue to resolve this.

Reply:
> On the other hand, Mbed CLI 1 doesn't understand `.o` unless major changes are made to the script and build system, therefore renaming to `.c` works.

I'm not sure about this. This file is only used by the MCUboot _signing script_, and we pass its path directly to the script ([here](https://github.com/ARMmbed/mbed-os/blob/45d9a22f0914ab1aae9530d0c56cbba523b87e99/tools/targets/ARM_MUSCA.py#L59)). It's not an actual source code file, so the build system should not care about or try to compile it as far as I understand.

Reply:
The content of the file is
```c
enum image_attributes {
 RE_SECURE_IMAGE_OFFSET = (0x0),
 RE_SECURE_IMAGE_MAX_SIZE = (0x60000),
 RE_NON_SECURE_IMAGE_OFFSET = ((0x0) + (0x60000)),
 RE_NON_SECURE_IMAGE_MAX_SIZE = (0x80000),
 RE_SIGN_BIN_SIZE = ((0x80000)),
};
```
It's not used anywhere in TF-M's C code. Instead, the signing script has regular expressions to match those sizes and offsets. In my opinion they could've named it `.txt` to avoid confusion here...

Reply:
What I'm saying is, the build system should not see/care about/compile those files, and the signing script will pick it up.

Reply:
Sorry its the `macro_parser.py`

Reply:
Have updated commit message for more clarity

Reply:
Thanks for the explanation and the commit message.